### ISSUE TITLE
log_user_ip config parameter is not used

### ISSUE THREAD

**Issue by teschnei** [DarkstarProject/darkstar#3139]
**_I have:_**

- [x] searched existing issues (http://github.com/darkstarproject/darkstar/issues/) to see if the issue I am posting has already been addressed or opened by another contributor
- [x] checked the commit log to see if my issue has been resolved since my server was last updated

**_Client Version_** (type `/ver` in game) **:**
n/a

**_Server Version_** (type `@revision` in game) **:**
a338f0d2b3cfac21f0b9a540766d23a9040462bb

**_Source Branch_** (master/stable) **:**
master

**_Additional Information_** (Steps to reproduce/Expected behavior) **:**

The log_user_ip parameter is supposed to disable logging ip when set to false.  The login server does not check if the value of the parameter is true, rather checks if the pointer is not null (which is always true).  Thus, it always logs IPs.

**Comment by teschnei** [comment]
@TeoTwawki

**Comment by TeoTwawki** [comment]
> The login server does not check if the value of the parameter is true, rather checks if the pointer is not null

https://github.com/DarkstarProject/darkstar/blob/01afbacdcf5b97cb64e82c519e0683eb1d43a5bd/src/login/lobby.cpp#L349

```c++
            if (login_config.log_user_ip == true)
            {
```

Pardon?

**Comment by teschnei** [comment]
update: the function that reads the value from the config file assigns a string to a bool, which will always be true

**Comment by TeoTwawki** [comment]
Forgot to close when you merged #3141

**Comment by teschnei** [comment]
it does it automatically for me when I push commits, guess not for pull reqs :\
